lib/nimble_options.ex

```diff
@@ -82,6 +82,8 @@ defmodule NimbleOptions do
 
     * `:non_empty_keyword_list` - A non-empty keyword list.
 
+    * `:map` - A map consisting of `:atom` keys.
```

Comment:
I think forcing this to have atom keys restricts us too much. We definitely might want to have a generic `:map` type in the future (like `t:map/0`). Can we call this `:atom_keyed_map`?

Reply:
My understanding is that this is used for when we specify the type with keys. So I would say saying `:map` is enough, as long as you also specify its keys? If you specify `:map` with no keys, then you either need to specify the key-value types or the keys. Would that make sense?

Reply:
Yeah but we don't support `:map` with key-value types, do we? I’m ok with either specifying `:keys` or specifying the type as `{:map, key_type, value_type}`.

Reply:
Hope I understood your suggestions, I have refactored to allow for both `:map` with `keys` specified or `{:map, key_type, value_type}`

Reply:
Additionally I added logic to generate the specs in the docs for both of these as well